test/core_test.exs

```diff
@@ -55,6 +55,80 @@ defmodule CoreTest do
     assert metrics_scrape =~ "http_request_total"
   end
 
+  test "initializes properly when configured via start_async" do
```

Comment:
I don't think we need to repeat all the tests for the default async behavior. That's already covered in all the other tests. You can just copy the test above and pass the new option to ensure it still starts fine. :)

Reply:
I can trim it down to just ensuring that it starts....but it did give me a little bit more confidence to assert that a sync start processes 1 less message from the mailbox on line 126.

Reply:
Thoughts?

Reply:
You can trim it down to just test that it still starts. I'm not worried about it :)

Reply:
Sounds good. It has been done!